docs/smart-contracts/views.md

```diff
@@ -2,86 +2,194 @@
+    sp.cast(b, sp.int)
+    if a > b:
+        return a
+    return b
+```
+
+This view returns a value from a big-map in storage:
+
+```python
+@sp.onchain_view
+def get(self, key):
+  valOpt = self.data.myBigmap.get_opt(key)
+  if valOpt.is_some():
+    return valOpt.unwrap_some()
+  else:
+    return ""
+```
+
+## Calling views in SmartPy
+
+In SmartPy tests, you can call views in the contract just like you call entrypoints.
+However, due to a limitation in SmartPy, if the view accepts multiple parameters, you must pass those parameters in a record.
+For example, to call the `get_larger` view in the previous example, use this code:
+
+```python
+viewResult = contract.get_larger(sp.record(a = 4, b = 5))
+scenario.verify(viewResult == 5)
+```
+
+To call a view in an entrypoint, pass the view name, target contract address, parameters, and return type to the `sp.view()` function, as in this example:
+
+```python
+@sp.entrypoint
+def callView(self, a, b):
+    sp.cast(a, sp.int)
+    sp.cast(b, sp.int)
+    viewResponseOpt = sp.view(
+        "get_larger", # Name of the view
+        sp.address("KT1K6kivc91rZoDeCqEWjH8YqDn3iz6iEZkj"), # Address of the contract
+        sp.record(a=a, b=b), # Parameters to pass
+        sp.int # Return type of the view
+    )
+    if viewResponseOpt.is_some():
+        self.data.myval = viewResponseOpt.unwrap_some()
+```
+
+If the view takes no parameters, pass `()` for the parameter:
+
+```python
+viewResponseOpt = sp.view(
+    "no_param_view", # Name of the view
+    sp.address("KT1K6kivc91rZoDeCqEWjH8YqDn3iz6iEZkj"), # Address of the contract
+    (), # No parameter
+    sp.int # Return type of the view
+)
+```
+
+## Calling views with Taquito
+
+Calling a view with Taquito is similar to calling entrypoints.
+When you create an object to represent the contract, its `contractViews` property has a method for each view, which you can call as in this example:
 
 ```javascript
-equalizeWith(destination)
-  destinationBalance = ledger.getBalance(destination)
-  totalBalance = ledger.getBalance(caller) + destinationBalance
-  targetBalance = totalBalance // 2
-  ledger.transfer(targetBalance - destinationBalance, destination)
+const viewContractAddress = "KT1K6kivc91rZoDeCqEWjH8YqDn3iz6iEZkj";
+const contract = await Tezos.wallet.at(viewContractAddress);
+const result = await contract.contractViews.get_larger({a: 2, b: 12})
+  .executeView({ viewCaller: viewContractAddress });
+console.log(result);
+```
+
+## Calling views with the Octez client
+
+To call a view with the Octez client, use the `run view` command, as in this example:
+
+```bash
+octez-client run view "get_larger" on contract "KT1Uh4MjPoaiFbyJyv8TcsZVpsbE2fNm9VKX" with input "Pair 4 5"
 ```
 
+If the view takes no parameters, you can pass Unit or omit the `with input`.
+<!-- TODO link to info on encoding param values -->
+
 ## Implementation details
 
 - Michelson: [Operations on views](https://tezos.gitlab.io/active/michelson.html#operations-on-views)
```

Comment:
Possibly also refer to this standalone page, which is much shorter that this one, but contains useful context about other kinds of views: <https://tezos.gitlab.io/active/views.html>

Reply:
In fact, the link you give doesn't even work anymore, so you should replace it with the page I suggested, but mind changing the link description.